pkg/migrations/sql/idempotent/base.sql

```diff
@@ -142,20 +142,19 @@ BEGIN
             CONTINUE;
         END IF;
 
-        IF SCHEMA_CATALOG.is_timescaledb_installed() AND SCHEMA_CATALOG.get_default_compression_setting() THEN
-            PERFORM SCHEMA_PROM.set_compression_on_metric_table(r.table_name, TRUE);
-        END IF;
-
         --do this before taking exclusive lock to minimize work after taking lock
         UPDATE SCHEMA_CATALOG.metric SET creation_completed = TRUE WHERE id = r.id;
 
         --we will need this lock for attaching the partition so take it now
-        --to avoid lock upgrade. Its critical to minimize work done while this
-        --lock is taken.
-        LOCK TABLE ONLY SCHEMA_CATALOG.series IN ACCESS EXCLUSIVE mode;
-
-        EXECUTE format($$
-           ALTER TABLE SCHEMA_CATALOG.series ATTACH PARTITION SCHEMA_DATA_SERIES.%1$I FOR VALUES IN (%2$L)
+        --This may not be strictly necessary but good
+        --to enforce lock ordering (parent->child) explcitly. Note:
+        --creating a table as a partition takes a stronger lock (access exclusive)
+        --so, attaching a partition is better
+        LOCK TABLE ONLY SCHEMA_CATALOG.series IN SHARE UPDATE EXCLUSIVE mode;
```

Comment:
Shouldn't this be `SHARE ROW EXCLUSIVE mode`?

Reply:
`SHARE UPDATE EXCLUSIVE` is the lock taken for attach partition